client/src/modules/Teams/Pages/Teams.tsx

```diff
@@ -0,0 +1,130 @@
+import { message, notification, Row } from 'antd';
+import { useMemo, useState } from 'react';
+import { PageLayout } from 'components/PageLayout';
+import { TeamsPageProps } from 'pages/course/teams';
+import {
+  JoinTeamModal,
+  MyTeamSection,
+  StudentsWithoutTeamSection,
+  TeamModal,
+  TeamsHeader,
+  TeamsSection,
+} from '../components';
+import { isActiveStudent, isCourseManager } from 'domain/user';
+import { useCopyToClipboard } from 'react-use';
+import { CreateTeamDto, JoinTeamDto, TeamApi, TeamDto } from 'api';
+import { showCreateTeamResultModal, showJoinTeamResultModal } from '../utils/showConfirmationModals';
+import { useDistribution } from '../hooks';
+
+const teamApi = new TeamApi();
+
+function Teams({ session, course, teamDistributionDetailed }: TeamsPageProps) {
+  const { distribution, loadDistribution, loading } = useDistribution(teamDistributionDetailed, course.id);
+
+  const [teamData, setTeamData] = useState<Partial<TeamDto> | null>(null);
+  const [showJoinTeamModal, setShowJoinTeamModal] = useState(false);
+
+  const [activeTab, setActiveTab] = useState('teams');
+
+  const [, copyToClipboard] = useCopyToClipboard();
+
+  const isManager = useMemo(() => isCourseManager(session, course.id), [session, course.id]);
+  const isStudent = useMemo(() => isActiveStudent(session, course.id), [session, course.id]);
+  const studentId = useMemo(() => session.courses[course.id]?.studentId, [session, course.id]);
+
+  const handleCreateTeam = () => {
+    setTeamData({});
```

Comment:
why empty object? looks weird and especially having the check for undefined below in jsx

Reply:
will be fixed in the next pr